File: C++/climbing-stairs.cpp
// This is a Dynamic Programming Problem
// The problem directly translates to the fibonacci problem
// F(n) = F(n-1) + F(n-2) where F(n) is the number of ways to reach nth stair

class Solution
{
public:
    int climbStairs(int n)
    {
        int a = 0;
        int b = 1;
        int c = 0;
        for (int i = 0; i < n; i++)
        {
            c = a + b;
            a = b;
            b = c;
        }
        return b;
    }
};

File: Java/Climbing-Stairs.java
class Solution {
    public int climbStairs(int n) {
        int prev1=1, prev2=2, cur=0;
        if(n<=2) return n;
        for(int i=3;i<=n;++i) {
            cur=prev1+prev2;
            prev1=prev2;
            prev2=cur;
        }
        return cur;
    }
}
